src/ctap/customization.rs

```diff
@@ -0,0 +1,247 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::ctap::data_formats::{CredentialProtectionPolicy, EnterpriseAttestationMode};
+
+/// ###########################################################################
+/// Constants for adjusting privacy and protection levels.
+/// ###########################################################################
+
+/// Changes the default level for the credProtect extension.
+///
+/// You can change this value to one of the following for more privacy:
+/// - CredentialProtectionPolicy::UserVerificationOptionalWithCredentialIdList
+/// - CredentialProtectionPolicy::UserVerificationRequired
+///
+/// UserVerificationOptionalWithCredentialIdList
+/// Resident credentials that are not in the allowList or excludeList can not be
+/// discovered without user verification.
+///
+/// UserVerificationRequired
+/// No resident credentials can be discovered without user verification.
+///
+/// This can improve privacy, but can make usage less comfortable.
+pub const DEFAULT_CRED_PROTECT: Option<CredentialProtectionPolicy> = None;
+
+/// Sets the intial minimum PIN length in code points.
+///
+/// # Invariant
+///
+/// - The minimum PIN length must be at least 4.
+/// - The minimum PIN length must be at most 63.
+/// - DEFAULT_MIN_PIN_LENGTH_RP_IDS must be non-empty if MAX_RP_IDS_LENGTH is 0.
+///
+/// Only the RP IDs listed in DEFAULT_MIN_PIN_LENGTH_RP_IDS are allowed to read
+/// the minimum PIN length with the minPinLength extension.
+/// Requiring longer PINs can help establish trust between users and relying
+/// parties. It makes user verification harder to break, but less convenient.
+/// NIST recommends at least 6-digit PINs in section 5.1.9.1:
+/// https://pages.nist.gov/800-63-3/sp800-63b.html
+///
+/// Reset reverts the minimum PIN length to this DEFAULT_MIN_PIN_LENGTH.
+pub const DEFAULT_MIN_PIN_LENGTH: u8 = 4;
+pub const DEFAULT_MIN_PIN_LENGTH_RP_IDS: &[&str] = &[];
```

Comment:
I split the documentation. Initial thought was that these two constants should be explained together, but I can split for consistency and relevance of documentation.

Reply:
It's not only consistency and relevance, it's the actual semantics of documentation. Previously only one item was documented. I think this comes from the fact that you expect users to read the file (in which case documentation looks a lot like comments if you don't know Rust) rather than read the generated documentation, which makes sense if they are expected to modify the file (and they don't really need to know Rust for that). So it would also makes sense for me if all documentation was switched to comments such that the file is meant to be read linearly and then multiple items can be described at once. I'm fine with both.